### PR TITLE
[WEBSITE-672] - Refresh the HOSTING automated message to reflect the plugin documentation pages

### PR DESCRIPTION
**Pull request by PierreBtz**
While following the process to host a new plugin I noticed that the documentation part was not up to date with the latest changes:

* the anchors of the two links didn't exist anymore
* the message didn't take into account the new preferred way to publish the doc

This is a proposal to refresh with a more up to date message.

### CODE REVIEW

File: src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java [hunk 1]
```diff
@@ -403,11 +403,10 @@ private void setupHosting(Channel channel, User sender, String hostingId, String
                     + "\n\nA JIRA component named " + forkTo + " has also been created with "
                     + defaultAssignee + " as the default assignee for issues."
                     + "\n\nPlease remove your original repository so that the jenkinsci repository "
-                    + "is the definitive source for the code. Also, please make sure you have "
-                    + "a wiki page set up with the following guidelines in mind: \n\n"
-                    + "* [Creating a wiki page|https://wiki.jenkins-ci.org/display/JENKINS/Hosting+Plugins#HostingPlugins-CreatingaWikipage]\n"
-                    + "* [Reference the wiki page from your plugin|https://wiki.jenkins-ci.org/display/JENKINS/Hosting+Plugins#HostingPlugins-AddingyourWikipagetoyourrepo]\n"
-                    + "\n\nYou will also need to do the following in order to push changes and release your plugin: \n\n"
+                    + "is the definitive source for the code. Also, please make sure you properly "
+                    + "follow the [host the plugin page|https://wiki.jenkins.io/display/JENKINS/Hosting+Plugins#HostingPlugins-Hostingthepluginpage] "
```
Comment (daniel-beck) [inline src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java]:
```suggestion
                    + "follow the [documentation on documenting your plugin|https://jenkins.io/doc/developer/publishing/documentation/] "
```

CC @oleg-nenashev

File: src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java [hunk 2]
```diff
@@ -403,11 +403,10 @@ private void setupHosting(Channel channel, User sender, String hostingId, String
...
+                    + "so that your plugin is correctly documented. \n\n"
+                    + "You will also need to do the following in order to push changes and release your plugin: \n\n"
                     + "* [Accept the invitation to the Jenkins CI Org on Github|https://github.com/jenkinsci]\n"
                     + "* [Request upload permission|https://wiki.jenkins-ci.org/display/JENKINS/Hosting+Plugins#HostingPlugins-Requestuploadpermissions]\n"
```
Comment (daniel-beck) [inline src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java]:
```suggestion
                    + "* [Request upload permission|https://github.com/jenkins-infra/repository-permissions-updater/#requesting-permissions]\n"
```

File: src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java [hunk 3]
```diff
@@ -403,11 +403,10 @@ private void setupHosting(Channel channel, User sender, String hostingId, String
...
                     + "* [Releasing your plugin|https://wiki.jenkins-ci.org/display/JENKINS/Hosting+Plugins#HostingPlugins-Releasingtojenkins-ci.org]\n"
```
Comment (daniel-beck) [inline src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java]:
```suggestion
                    + "* [Releasing your plugin|https://jenkins.io/doc/developer/publishing/releasing/]\n"
```

**Comment by oleg-nenashev** [comment]
Is it a Hacktoberfest pull request just in case? Just to label it properly on our side. See [Hacktoberfest in Jenkins](https://jenkins.io/events/hacktoberfest). If you do not participate in it, please consider doing so. Your pull requests are much appreciated 👍

**Comment by slide** [comment]
LGTM 👍

**Comment by PierreBtz** [comment]
@oleg-nenashev here you go https://issues.jenkins-ci.org/browse/WEBSITE-672 do you need me to rewrite my commit message with this for tracking?

**Comment by oleg-nenashev** [comment]
No need, thanks! Can be tracked from the PR

**Comment by slide** [comment]
I don't have access to merge in this repo, can someone else do it?

**Comment by oleg-nenashev** [comment]
I will get it deployed. Fixing up docs and the release cycle in the process

**Comment by oleg-nenashev** [comment]
https://github.com/jenkins-infra/jenkins-infra/pull/1381 for deployment

**Pull request merged**